test/utils.cpp

```diff
@@ -814,6 +814,31 @@ TEST_CASE("getcwd() returns current directory of the process", "[utils]")
 		REQUIRE(datadir.substr(datadir.length() - subdir.length()) ==
 			subdir);
 	}
+
+	SECTION("Invalid current directories")
+	{
+		const std::string maindir = utils::getcwd();
+		// Create a temporary directory, change to it and delete it. This leads
```

Comment:
We have a helper class for that, called `TestHelpers::TempDir`. It creates the directory and removes it when the test ends (regardless of test outcome), and you can get the path as a `std::string` using `get_path()`.

There's nothing similar for `chdir`, though.

Reply:
I added something similar for `chdir` and rewrote the tests accordingly.